third_party/nvidia/lib/TritonNVIDIAGPUToLLVM/LoadStoreOpToLLVM.cpp

```diff
@@ -632,6 +649,27 @@ struct AtomicRMWOpConversion
       : ConvertOpToLLVMPattern<triton::AtomicRMWOp>(converter, benefit),
         LoadStoreConversionBase(targetInfo, axisAnalysisPass) {}
 
+  bool supportsVectorized(Operation *moduleOp, RMWOp opType,
+                          Type elementType) const {
+    // vectorized atomics are only supported on hopper,
+    // and only for specific atomic ops (add, min, max).
+    // Note that "packed types" like f16x2 are supported sm60+.
+    auto computeCapability = getNVIDIAComputeCapability(moduleOp);
+    if (computeCapability < 90) {
+      return false;
+    }
+
+    switch (opType) {
+    case RMWOp::FADD:
+      return elementType.isF16() || elementType.isBF16() || elementType.isF32();
+    case RMWOp::MAX:
+    case RMWOp::MIN:
+      return elementType.isF16() || elementType.isBF16();
```

Comment:
Note: atomic_max doesn't support f16 right now due to frontend checks

https://github.com/triton-lang/triton/blob/1064b598385c49f03fbc73f6839f578146beb4e4/python/triton/language/semantic.py#L1304-L1305

Reply:
So you meant there's no problem supporting it in the backend, right?
Why not we remove the check?

Reply:
Sure - removed these cases